### PR TITLE
set all unset ENV variables to nil

### PR DESCRIPTION
**Pull request by h-artzi**
### What does this PR do?
This will ensure there are no issues with ENV variables that are
accidentally set to empty strings

### What ticket does this PR close?
Connected to #1841 

#### Change log
- [X] The CHANGELOG has been updated, or
- [] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [X] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [X] This PR does not require updating any documentation

### CODE REVIEW

File: bin/conjur-cli.rb [hunk 1]
```diff
@@ -59,6 +59,12 @@ def test_select
   c.action do |global_options,options,args|
     account = options[:account]
 
+    # Sets all the `unset` Environment Variables to nil. This ensures that nil
+    # checks are sufficient to verify the usage of an environment variable
+    ENV.each_pair do |(k,v)|
+      ENV[k] = nil if v =~ /^ *$/ # is all whitespace
```
Comment (jonahx) [inline bin/conjur-cli.rb:65]: use `/^\s*$/` for any whitespace character.  as is, it's only spaces.

File: CHANGELOG.md [hunk 2]
```diff
@@ -19,6 +19,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
   [cyberark/conjur#151](https://github.com/cyberark/conjur/issues/151)
 - [Documentation](./UPGRADING.md) explaining how to upgrade a Conjur server deployed in a
   Docker Compose environment. [cyberark/conjur#1528](https://github.com/cyberark/conjur/issues/1528), [cyberark/conjur#1584](https://github.com/cyberark/conjur/issues/1584)
+- Blank environment variables are set nil when Conjur is started. [cyberark/conjur#1841](https://github.com/cyberark/conjur/issues/1841)
```
Comment (izgeri) [inline CHANGELOG.md:22]: can this be more clear about the user impact / benefit of this change?

File: bin/conjur-cli.rb [hunk 3]
```diff
@@ -59,6 +59,12 @@ def test_select
   c.action do |global_options,options,args|
     account = options[:account]
 
+    # Sets all the `unset` Environment Variables to nil. This ensures that nil
+    # checks are sufficient to verify the usage of an environment variable
+    ENV.each_pair do |(k,v)|
+      ENV[k] = nil if v =~ /^\s*$/ # is all whitespace
+    end
+
```
Comment (micahlee) [inline bin/conjur-cli.rb:67]: @h-artzi Is this only relevant for OSS?

DAP doesn't use `conjurctl` to run the app, so this wouldn't work there.

**Comment by h-artzi** [comment]
@jonahx this [resource](https://guides.rubyonrails.org/v2.3/configuring.html) was used to determine the usage of `preinitializer.rb`

File: CHANGELOG.md [hunk 4]
```diff
@@ -19,6 +19,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
   [cyberark/conjur#151](https://github.com/cyberark/conjur/issues/151)
 - [Documentation](./UPGRADING.md) explaining how to upgrade a Conjur server deployed in a
   Docker Compose environment. [cyberark/conjur#1528](https://github.com/cyberark/conjur/issues/1528), [cyberark/conjur#1584](https://github.com/cyberark/conjur/issues/1584)
+- When Conjur starts, we convert blank environment variables to nil. This makes blank environment values behave like
+  non-existent ones. [cyberark/conjur#1841](https://github.com/cyberark/conjur/issues/1841)
```
[the remark this answers is not in the document]
Comment (h-artzi) [reply CHANGELOG.md:23]: I think that sounds good, I would love to talk about the depth we want our changelog entries to be.
1. Should they cover the impact on the user experience?
2. Should they cover the impact on the developer?
3. Or combination of them both?

[the remark this answers is not in the document]
Comment (micahlee) [reply CHANGELOG.md:23]: Yeah, it's certainly an area open to discussion. For myself, I've typically thought about the changelog as just **"what changed?"** I've realized however, based on various roles as the consumer of a changelog, it should also really answer the question, **"Why does this matter?"** or **"Does this matter to me?"**

I imagine whether it impacts the end-user, the developer, or both will vary based on what the change is. The key thing though is how do you know, unless the changelog tells you?

**Review by micahlee — DISMISSED** [review]
LGTM, thanks!

**Review by micahlee — APPROVED** [review]

**Pull request merged**